src/components/core/toast/store.js

```diff
@@ -89,17 +89,23 @@ const toastQueueStore = createStore({
   maxVisible: MAX_VISIBLE_TOASTS,
 });
 
-// Counter to ensure unique IDs
-let idCounter = 0;
+function* infiniteIdGenerator() {
+  let id = 0;
+
+  while (true) {
+    yield id++;
+  }
+}
+
+const idGenerator = infiniteIdGenerator();
 
 /**
  * Add a new toast to the queue
  * @param {Partial<ToastState>} newToast - Toast to add
  */
 const addToast = (newToast) => {
-  // Generate unique ID using counter + timestamp
   const now = Date.now();
-  const uniqueId = now + ++idCounter;
+  const uniqueId = idGenerator.next().value;
```

Comment:
You created the uniqueId by concatenating a timestamp with an increment number. We could still implement it as a generator. What do you think?

Reply:
Initially I used Date.now() as ID but added an incrementing counter to avoid collisions. After your generator suggestion, I realized that for toast's short-lived nature, simple incrementing IDs are sufficient and more readable. Since toasts disappear on page reload anyway, cross-session uniqueness isn't needed.
I'd prefer to keep the current approach for simplicity unless you see specific benefits of integrating timestamp into the ID generation?

Reply:
I'm comfortable with both options; I just need to confirm your approach. Thank you for your response.